### PR TITLE
8236225: Remove expired flags in JDK 17 

### PR DESCRIPTION
**Pull request by dholmes-ora**
8236225: Remove expired flags in JDK 17 
8256717: Expire the long term obsoleted VM flags 

The newly expired flags in 17 are removed from the flag table, as are all of the long term obsolete flags that had no explicit expiration release.

Updated any remaining references to the obsolete flags that are now expired.

Updated the man page to move flags from obsolete section to expired (removed) section. Note the manpage also shows changes to the illegal-access flag, this is a change that was made to the closed manpage source for JDK 16 but has not yet been pushed into the open java.1 version. This part of the change can be ignored.

Testing: tiers 1-3 and local testing of modified tests. I'm also checking for any JCK tests that may still use the legacy obsolete flags.

Thanks,
David

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issues
 * [JDK-8236225](https://bugs.openjdk.java.net/browse/JDK-8236225): Remove expired flags in JDK 17
 * [JDK-8256717](https://bugs.openjdk.java.net/browse/JDK-8256717): Expire the long term obsoleted VM flags

### Reviewers
 * [Coleen Phillimore](https://openjdk.java.net/census#coleenp) (@coleenp - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1773/head:pull/1773`
`$ git checkout pull/1773`

### CODE REVIEW

File: src/hotspot/share/runtime/arguments.cpp [hunk 1]
```diff
@@ -537,42 +536,9 @@ static SpecialFlag const special_jvm_flags[] = {
   { "TLABStats",                    JDK_Version::jdk(12), JDK_Version::undefined(), JDK_Version::undefined() },
 
   // -------------- Obsolete Flags - sorted by expired_in --------------
-  { "PermSize",                      JDK_Version::undefined(), JDK_Version::jdk(8),  JDK_Version::undefined() },
```
Comment (coleenp) [inline src/hotspot/share/runtime/arguments.cpp:540]: No objecting, but it doesn't look like we told them when it would become unrecognized.

**Review by coleenp — APPROVED** [review]
LGTM!

**Review by vnkozlov — APPROVED** [review]

File: src/hotspot/share/runtime/arguments.cpp (hunk 1 above)
Comment (dholmes-ora) [reply src/hotspot/share/runtime/arguments.cpp:540]: We don't report that value in the warning we only report the release in which it became obsolete. e.g.:

Java HotSpot(TM) 64-Bit Server VM warning: Ignoring option PermSize; support was removed in 8.0

So to all intents and purposes you can read all these undefined() entries as if they said jdk(17).

Thanks for the review.

**Pull request closed without merging**